Extend caching docu on cache filters (#369)
SAP/cloud-sdk: docs/java/features/resilience/caching.mdx

:::

> If you need fine-grained influence on which cache entries to remove, you may write your custom `CacheFilter`.

Reviewer: ```suggestion
If you need fine-grained control over invalidating of cache entries, you may write your custom `CacheFilter`.
```

> If you need fine-grained control over invalidating of cache entries, you may write your custom `CacheFilter`.

Reviewer 2: I think we should stick with one term for "clearing"/"invalidating" cache entries.

```suggestion
If you need fine-grained control over which cache entries to clear, you may write your custom `CacheFilter`.
```

:::

If you need fine-grained influence on which cache entries to remove, you may write your custom `CacheFilter`.

```java
// CacheFilter which matches all cache entries belonging to tenant "tenant1"

> CacheFilter tenantFilter = ( configuration, cacheKey, cacheEntry ) -> cacheKey.getTenantId().isDefined() && cacheKey.getTenantId().get().equals("tenant1");

Author: I know that this can be abbreviated with `keyMatchesTenant(...)`, but I find this good enough to illustrate how to create your own custom filter.
